First cut at YARA docs.

wxsBSD opened the pull request:
This is in need of some work, but might as well toss it up and get more eyes on it.

osqueryer commented:
Can one of the osquery admins reply with "ok to test" to kick off a build...

marpaia commented on docs/wiki/deployment/yara.md:
```diff
@@ -0,0 +1,138 @@
+# Introduction
+
+I recently put [YARA](https://github.com/plusvic/yara) inside [osquery](https://osquery.io) and thought I would provide some details on how to use it. There are two YARA related tables in osquery, which serve very different purposes. The first table, called **yara_events**, uses osquery's pub-sub framework to monitor for filesystem changes and will execute YARA when a file change event fires. The ...
```
I would cut this whole first sentence. The docs should read as third-person reference, rather than first-person.

marpaia replied:
Maybe make "pub-sub framework" a link to http://osquery.readthedocs.org/en/latest/development/pubsub-framework/

wxsBSD replied:
I was thinking of that when I woke up. I forgot about the introduction entirely. :)

marpaia commented on docs/wiki/deployment/yara.md:
```diff
@@ -0,0 +1,138 @@
+
+# Configuration
+
+The configuration for osquery is simple. Here is an example config (I've left out the non-relevant parts):
```
I would include the minimum viable non-relevant parts (which I think should just be braces around this) and get rid of the qualifying note there.

marpaia commented on docs/wiki/deployment/yara.md:
```diff
@@ -0,0 +1,138 @@
+  "yara": {
+    "signatures": {
+      // Each key is an arbitrary group name to give the signatures listed
+      "sig_group_1": [ "/Users/wxs/sigs/foo.sig", "/Users/wxs/sigs/bar.sig" ],
```
we should take this opportunity to set a precedent for where yara signatures should be installed on the filesystem. Maybe make all of these paths `/etc/osquery/yara/foo.sig` and such?

wxsBSD replied:
Right now the YARA table will check in ```/var/osquery``` if the sigfile is not an absolute path. I can change it to be ```/etc/osquery/yara/``` if you would like? I only went with ```/var/osquery``` because it was mentioned in an earlier PR. I will also make the config parser have the same logic.

marpaia commented on docs/wiki/deployment/yara.md:
````diff
@@ -0,0 +1,138 @@
+Using the configuration above you can see it in action. While osquery is running I executed *touch /Users/wxs/tmp/foo* in another terminal. Here is the relevant queries to show what happened:
+
+```
+osquery> select * from file_changes;
````
we're going to be changing this to `file_events`, as per #1049

marpaia commented on docs/wiki/deployment/yara.md:
````diff
@@ -0,0 +1,138 @@
+osquery> 
+```
+
+The **file_changes** table recorded that a file named */Users/wxs/tmp/foo* was created with the corresponding hashes and a timestamp.
````
#1049 here too

marpaia commented on docs/wiki/deployment/yara.md:
```diff
@@ -0,0 +1,138 @@
+
+The **yara** table is used for on-demand scanning. With this table you can arbitrarily YARA scan any available file on the filesystem with any available signature files or signature group from the configuration. In order to scan the table must be given a constraint which says where to scan and what to scan with.
+
+In order to determine where to scan the table accepts either a *path* or a *pattern* constraint. The *path* constraint must be a full path to a single file. There is no expansion or recursion with this constraint. The *pattern* constraint follows the same [wildcard rules](http://osquery.readthedocs.org/en/latest/development/wildcard-rules/index.html) mentioned before.
```
Do you think it's worth us adding stand-alone documentation somewhere on the logic behind the whole `pattern = "some/pattern"` thing? People in chat have called it out for being a bit confusing.

wxsBSD replied:
If people have said it's confusing it could use some stand-alone docs for sure.

marpaia commented:
These look really hot. Thanks so much for doing this, @wxsBSD.

wxsBSD commented:
I believe I've addressed all concerns, which looking back were rather obvious given that this went from a gist to get the word out to a doc PR pretty quickly.

theopolis commented:
Cool! Will land first thing tomorrow morning!
